Add RFP-013 reflexive stablecoin and RFP-014 liquidation engine (#25)
logos-co/rfp: RFPs/RFP-014-liquidation-auction-engine.md

3. Auctions support buying partial collateral lots, preventing large positions from becoming stuck.

4. When collateral auctions don't raise enough to cover debt plus penalty, the shortfall is handled via a separate debt auction mechanism that restores system balance.

5. When the protocol accumulates excess stablecoin above a threshold, surplus auctions sell it for the reference token (which is burned).

6. Auctions follow strict state transitions with no reversals. All state changes are atomic and verifiable.

7. The system integrates with its host CDP protocol via a well-defined interface. Each product deploys its own configured instance of the program.

#### Usability

1. Build the program using the SPEL framework, which generates the IDL and client code from the program definition.

2. Provide a Logos mini-app GUI with local build instructions, downloadable assets, and loadable in Logos app (Basecamp) via git repo.

3. Provide a CLI that covers core functionality of the program. The CLI may have fewer features than the GUI mini-app but must support all essential operations.

> 4. Tools display undercollateralized positions with estimated liquidator incentives, enabling efficient bot operation.

Reviewer: Isn't this a "traceability" layer on top of the liquidation program? Why do we want this to be part of the liquidation engine itself?

Author: Liquidator bots need to know which positions are underwater and whether it's profitable to liquidate them. If the program doesn't expose this data, every liquidator has to build their own position scanning and reward estimation from raw account state. That's a duplicated effort across every bot operator. 

Therefore, I'd consider this a usability requirement for the tools (CLI/mini-app/SDK), not an on-chain traceability layer. It's just displaying the current state of underwater positions and their liquidation incentives. There's no history tracking or audit trail here, it's static data. The program itself just needs to make the data queryable.

5. Active auctions display collateral type, amount, current discount, raised amount vs target, and time remaining. Users can bid with clear previews of received collateral and costs.

> 6. SDK supports private account operations via the deshield-interact-reshield pattern.

Reviewer 2: Same as RFP-013: the SDK is referenced as a constraint but not required as a deliverable. Suggest:

```suggestion
6. Provide an SDK that can be used to build Logos modules for interacting with the program. The SDK must support private account operations via the deshield-interact-reshield pattern.
```

#### Reliability

1. Auction operations are non-reentrant. State transitions are ordered to prevent manipulation.

> 2. Liquidation uses time-weighted or confidence-adjusted prices, not raw spot. If price feeds are stale, liquidations pause for that collateral type.

Reviewer: Should we add a dependency on the oracle's RFP?

Author: We don't have this RPF yet. It's listed as a hard blocker ("Price feed") in the Platform Dependencies section.

3. Multiple liquidations and auctions can occur simultaneously without interference. Each auction maintains isolated state.

4. If a price feed fails for one collateral type, only that type's liquidations pause. Other operations continue unaffected.

> 5. Critical invariants: seized collateral equals collateral in auction; debt shortfalls are properly accounted; surplus and debt auctions maintain system balance.

Reviewer 2: Two distinct testable statements — suggest splitting and renumbering R5 accordingly:

```suggestion
4. If a price feed fails for one collateral type, only that type's liquidations pause.

5. Other collateral types and non-liquidation operations continue unaffected.

6. Critical invariants: seized collateral equals collateral in auction; debt shortfalls are properly accounted; surplus and debt auctions maintain system balance.
```

#### Performance

1. Liquidation transactions complete within one block.

2. The system supports many concurrent auctions without exceeding compute limits.

3. Bid transactions complete within one block.

> #### Supportability

Reviewer: I've already seen this Section in a few other RFPs. Shouldn't we create a "global FURPS" and "RFP-specific FURPS" to avoid having to rewrite this for each RFP?
This also reduces the amount of work required if we want/need to update an entry of this list.

Author: Good observation. I guess the reason we haven't done it is that some future RFPs may target different parts of the stack (tooling, infrastructure, storage, messaging) where the current boilerplate (deploy on testnet, SPEL IDL, mini-app, etc.) wouldn't apply.

1. The program is deployed and tested on LEZ devnet/testnet.

> 2. End-to-end integration tests run against a LEZ sequencer (standalone mode) and are included in CI.

Reviewer 2: Suggest requiring real proof generation in CI:

```suggestion
2. End-to-end integration tests run against a LEZ sequencer (standalone mode) with `RISC0_DEV_MODE=0` and are included in CI.
```
